### PR TITLE
L2 revamp

### PR DESCRIPTION
**Pull request by corwintines**
## Description

[L2 hub](https://deploy-preview-14251--ethereumorg.netlify.app/en/layer-2)
[L2 learn](https://deploy-preview-14251--ethereumorg.netlify.app/en/layer-2/learn)
[L2 networks](https://deploy-preview-14251--ethereumorg.netlify.app/en/layer-2/networks)
[Find walletes](https://deploy-preview-14251--ethereumorg.netlify.app/en/wallets/find-wallet)

## Related Issue

fixes https://github.com/ethereum/ethereum-org-website/issues/13062

### CODE REVIEW

File: src/components/Layer2NetworksTable/NetworkUsageChart.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
+"use client"
```
Comment (pettinarip) [inline src/components/Layer2NetworksTable/NetworkUsageChart.tsx:1]:
```suggestion
```

File: src/components/Layer2NetworksTable/NetworkUsageChart.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+import { Bar, BarChart, CartesianGrid, LabelList, YAxis } from "recharts"
```
Comment (pettinarip) [inline src/components/Layer2NetworksTable/NetworkUsageChart.tsx:3]: We will need to decide later if we want to keep using this or `chart.js`.

File: src/components/FindWalletProductTable/hooks/useWalletColumns.tsx
Comment (pettinarip) [inline src/components/FindWalletProductTable/hooks/useWalletColumns.tsx:1]: Not sure where this is coming from but there is a mismatch in the expected html
image

File: src/components/Callout.tsx
Comment (pettinarip) [inline src/components/Callout.tsx:1]: In the gas and some other pages (the ones using the Callout) have spacing issues
image

File: src/components/icons/layer-2/MaturingIcon.tsx
Comment (pettinarip) [inline src/components/icons/layer-2/MaturingIcon.tsx:1]: This icon is not switching in dark mode

image

File: public/images/layer-2/linea.png
Comment (pettinarip) [inline public/images/layer-2/linea.png:1]: Would be nice if we can add a dark mode version of this logo.
image

**Review by pettinarip — COMMENTED** [review]
Nice job @corwintines! 🎉  Lot of work done in this PR.

File: src/pages/layer-2/networks.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+}) satisfies GetStaticProps<BasePageProps>
+
+const Layer2Networks = ({ layer2Data, locale, mainnetData }) => {
+  const { pathname } = useRouter()
+
+  const heroProps: ContentHeroProps = {
+    breadcrumbs: { slug: pathname, startDepth: 1 },
+    heroImg: "/images/layer-2/learn-hero.png",
+    blurDataURL: "/images/layer-2/learn-hero.png",
+    title: "Choose network",
```
Comment (wackerow) [inline src/pages/layer-2/networks.tsx:136]: "Choose network" feels restrictive to me as the title here... feel like you choose one and that's all, but I feel like "Explore networks" fits what we're trying to present better... @konopkja Any thoughts on this from SEO perspective?

File: src/pages/layer-2/index.tsx [hunk 4]
```diff
@@ -0,0 +1,630 @@
...
+          <h2>The network of networks</h2>
+          <p className="font-md">
+            Ethereum&apos;s strength and security provides a platform for other
+            networks to build upon. With a single account, everything is
+            compatible and connects seamlessly.
+          </p>
+          <div className="relative m-auto h-[275px] w-[275px] sm:h-[375px] sm:w-[375px]">
+            {/* Outer ring */}
+            <div className="absolute inset-0 rounded-full border border-dashed border-body-medium"></div>
+            <div className="absolute inset-0 animate-spin-30 rounded-full">
```
Comment (wackerow) [inline src/pages/layer-2/index.tsx:232]: image
image

Just noting some of these are hard to see in light mode, and others in dark. Perhaps we force a circular background color here?

File: src/pages/layer-2/index.tsx [hunk 5]
```diff
@@ -0,0 +1,630 @@
...
+    {
+      title: "$0.01 fees",
+      description:
+        "You can trade, send money globally, or use  applications without worrying about high costs.",
+      emoji: ":money_with_wings:",
+    },
+    {
+      title: "Near instant transactions",
+      description:
+        "Whether you are making a quick payment or engaging in decentralized finance (DeFi), all transactions take only few seconds. ",
```
Comment (wackerow) [inline src/pages/layer-2/index.tsx:119]:
```suggestion
        "Whether you are making a quick payment or engaging in decentralized finance (DeFi), all transactions take only a few seconds. ",
```

File: src/pages/layer-2/index.tsx [hunk 6]
```diff
@@ -0,0 +1,630 @@
...
+      emoji: ":closed_lock_with_key:",
+    },
+    {
+      title: "Backed by Ethereum",
+      description:
+        "Ethereum’s time proven and decentralized blockchain functions as the settlement layer for other newer networks.",
```
Comment (wackerow) [inline src/pages/layer-2/index.tsx:125]:
```suggestion
        "Ethereum’s time-proven and decentralized blockchain functions as the settlement layer for other newer networks.",
```

File: src/pages/layer-2/index.tsx [hunk 7]
```diff
@@ -0,0 +1,630 @@
...
+        </div>
+      </div>
+
+      {/* TODO: Setup for translation */}
+      <div
+        id="layer-2-callout-cards"
+        className="flex w-full flex-col gap-16 px-8 py-8 md:flex-row"
+      >
+        <Callout
+          image={ExploreImage}
```
Comment (wackerow) [inline src/pages/layer-2/index.tsx:585]: image

This image gets cut off

Comment (wackerow) [reply src/pages/layer-2/index.tsx:585]: I see we're reusing this from the /networks/ hero.. I wonder if we have another option that isn't chopped at the top

File: src/intl/en/page-layer-2-learn.json [hunk 8]
```diff
@@ -0,0 +1,55 @@
+{
+  "page-layer-2-learn-meta-title": "What is layer 2",
+  "page-layer-2-learn-title": "What is layer 2",
+  "page-layer-2-learn-description": "Scaling Ethereum for mass adoption",
+  "page-layer-2-learn-button-1-label": "What is layer 2",
```
Comment (wackerow) [inline src/intl/en/page-layer-2-learn.json:5]: Same as above, any reason we left the "?" off here?

**Review by wackerow — COMMENTED** [review]
Left some initial thoughts... over all this looks fantastic, great job!

File: src/pages/layer-2/index.tsx (hunk 7 above)
Comment (corwintines) [reply src/pages/layer-2/index.tsx:585]: Its not that its cut off, its just how that image is. I think we could pick another image if we want, ill bring it up with Jakub.

File: src/pages/layer-2/index.tsx (hunk 4 above)
Comment (corwintines) [reply src/pages/layer-2/index.tsx:232]: Updated the logos and rounded effect

File: src/components/Layer2NetworksTable/NetworkUsageChart.tsx (hunk 2 above)
Comment (corwintines) [reply src/components/Layer2NetworksTable/NetworkUsageChart.tsx]: Makes sense. I was just implementing the shadcn stuff, which uses recharts. I can refactor if you want.

File: src/intl/en/page-layer-2.json
[the remark this answers is not in the document]
Comment (corwintines) [reply src/intl/en/page-layer-2.json:1]: I will be using this when I set translations up, so just leaving it in for now.

File: src/styles/semantic-tokens.css [hunk 9]
```diff
@@ -75,6 +75,12 @@
       transparent 100%
     );
 
+    --linear-gradient-step-1: linear-gradient(
```
[the remark this answers is not in the document]
Comment (corwintines) [reply src/styles/semantic-tokens.css:78]: Nope, removing!

File: src/lib/api/fetchEthereumTVL.ts [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import { MetricReturnData } from "../types"
+
+export const fetchEthereumTVL = async (): Promise<MetricReturnData> => {
```
[the remark this answers is not in the document]
Comment (corwintines) [reply src/lib/api/fetchEthereumTVL.ts:3]: I should rename this to marketcap and not TVL. The TVL one has that you mentioned has more to do with defi TVL, where I mean marketcap here. Will update.

File: src/components/Callout.tsx
Comment (corwintines) [reply src/components/Callout.tsx:1]: Ah good catch, will fix the margins here

File: src/pages/layer-2/index.tsx (hunk 7 above)
Comment (wackerow) [reply src/pages/layer-2/index.tsx:581]: Right right, it works as a hero where it's in a box, but isn't great for this spot imo, cc @konopkja

Comment (corwintines) [reply src/pages/layer-2/index.tsx:581]: core

Maybe this one would be ok for now, and we can always change later?

File: public/content/translations/cs/developers/docs/scaling/optimistic-rollups/index.md
Comment (wackerow) [inline public/content/translations/cs/developers/docs/scaling/optimistic-rollups/index.md:1]: cc: @lukassim We'll just want to make sure the deletions from the two `developers/docs/scaling/**/index.md` files here (and all translations) are propagated in Crowdin

File: src/pages/layer-2/index.tsx [hunk 11]
```diff
@@ -0,0 +1,630 @@
...
+          </div>
+        </div>
+      </div>
+
+      {/* TODO: Setup for translation */}
+      <div id="layer-2-stats-box" className="w-full px-8 py-9">
+        <div className="m-auto max-w-[992px] py-9">
+          <div className="flex flex-col gap-8 border border-body-light p-8 md:flex-row md:gap-14">
+            <div className="flex-1">
+              <div className="max-w-[224px]">
```
[the remark this answers is not in the document]
Comment (corwintines) [reply src/pages/layer-2/index.tsx:181]: Any thoughts about changing this stats design to whats on the homepage @konopkja?

File: src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx [hunk 12]
```diff
@@ -0,0 +1,241 @@
...
+            Networks showing <strong>({table.options.data.length})</strong>
+          </p>
+        </TableHead>
+      )
+    },
+    cell: ({ table, row }) => {
+      const meta = table.options.meta as TableMeta
+      return (
+        <TableCell
+          className={`flex flex-1 flex-row items-center justify-between lg:table-cell ${row.original.cantExpand ? "border-b-4" : ""}`}
```
Comment (wackerow) [inline src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx:32]: Can use `cn` here to merge these
```suggestion
          className={cn(
            "flex flex-1 flex-row items-center justify-between lg:table-cell",
            row.original.cantExpand && "border-b-4"
          )}
```

File: src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx [hunk 13]
```diff
@@ -0,0 +1,241 @@
...
+import { ColumnDef } from "@tanstack/react-table"
+
+import { ExtendedRollup, Lang } from "@/lib/types"
+
+import { TableMeta } from "@/components/DataTable"
+import { TwImage } from "@/components/Image"
+import NetworkMaturityTooltip from "@/components/Layer2NetworksTable/NetworkMaturityTooltip"
+import InlineLink from "@/components/Link"
+import Tooltip from "@/components/Tooltip"
+import { TableCell, TableHead } from "@/components/ui/Table"
```
Comment (wackerow) [inline src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx:14]: With suggestion below

```suggestion
import { TableCell, TableHead } from "@/components/ui/Table"


import { cn } from "@/lib/utils/cn"
```

File: src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx [hunk 14]
```diff
@@ -0,0 +1,241 @@
...
+          </span>
+        </p>
+      </TableHead>
+    ),
+    cell: ({ table, row }) => {
+      const meta = table.options.meta as TableMeta
+
+      return (
+        <TableCell
+          className={`hidden w-[145px] px-0 text-end lg:table-cell ${row.original.cantExpand ? "border-b-4" : ""}`}
```
Comment (wackerow) [inline src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx:122]:
```suggestion
          className={cn(
            "hidden w-[145px] px-0 text-end lg:table-cell",
            row.original.cantExpand && "border-b-4"
          )}
```

File: src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx [hunk 15]
```diff
@@ -0,0 +1,241 @@
...
+            </Tooltip>
+          </span>
+        </p>
+      </TableHead>
+    ),
+    cell: ({ table, row }) => {
+      const meta = table.options.meta as TableMeta
+      return (
+        <TableCell
+          className={`hidden w-[120px] px-0 text-end lg:table-cell ${row.original.cantExpand ? "border-b-4" : ""}`}
```
Comment (wackerow) [inline src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx:162]:
```suggestion
          className={cn(
            "hidden w-[120px] px-0 text-end lg:table-cell",
            row.original.cantExpand && "border-b-4"
          )}
```

File: src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx [hunk 16]
```diff
@@ -0,0 +1,241 @@
...
+              <MdInfoOutline className="translate-y-0.5" />
+            </Tooltip>
+          </span>
+        </p>
+      </TableHead>
+    ),
+    cell: ({ row }) => {
+      return (
+        <TableCell
+          className={`hidden w-[145px] px-0 text-center lg:table-cell ${row.original.cantExpand ? "border-b-4" : ""}`}
```
Comment (wackerow) [inline src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx:208]:
```suggestion
          className={cn(
            "hidden w-[145px] px-0 text-center lg:table-cell",
            row.original.cantExpand && "border-b-4"
          )}
```

File: src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx [hunk 17]
```diff
@@ -0,0 +1,241 @@
...
+    },
+  },
+  {
+    id: "dropdown",
+    header: () => <TableHead className="hidden w-12 lg:table-cell" />,
+    cell: ({ row }) => {
+      if (row.original.cantExpand)
+        return (
+          <TableCell
+            className={`hidden w-12 lg:table-cell ${row.original.cantExpand ? "border-b-4" : ""}`}
```
Comment (wackerow) [inline src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx:222]:
```suggestion
            className={cn(
              "hidden w-12 lg:table-cell",
              row.original.cantExpand ? "border-b-4" : "",
            )}
```

File: src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx [hunk 18]
```diff
@@ -0,0 +1,241 @@
...
+          />
+        )
+
+      return (
+        <TableCell
+          className={`hidden w-12 lg:table-cell ${row.original.cantExpand ? "border-b-4" : ""}`}
```
Comment (wackerow) [inline src/components/Layer2NetworksTable/hooks/useNetworkColumns.tsx:228]:
```suggestion
          className={cn(
            "hidden w-12 lg:table-cell",
            row.original.cantExpand ? "border-b-4" : "",
          )}
```

**Review by wackerow — COMMENTED** [review]
Can use `cn` to merge all of these safely

**Comment by wackerow** [comment]
image

360px width:
image

cc: @konopkja Are we sure we need/want to shrink the icons this much? They're quite small dna we seem to have plenty of space... or at the very least, I think we could make them larger at a lower breakpoint.

File: src/pages/layer-2/index.tsx (hunk 11 above)
[the remark this answers is not in the document]
Comment (konopkja) [reply src/pages/layer-2/index.tsx:181]: not opposed, but how would it look like with 2 stats? would there be no bottom line?

File: src/pages/layer-2/networks.tsx (hunk 3 above)
Comment (konopkja) [reply src/pages/layer-2/networks.tsx]: fine with explore

File: src/pages/layer-2/index.tsx (hunk 7 above)
Comment (konopkja) [reply src/pages/layer-2/index.tsx:585]: hmm i guess its ok to change though will be a little inconsistent with the hero

**Comment by konopkja** [comment]
> image
> 360px width: image
> 
> cc: @konopkja Are we sure we need/want to shrink the icons this much? They're quite small dna we seem to have plenty of space... or at the very least, I think we could make them larger at a lower breakpoint.

they should be bigger, ye on desktop especially

**Review by pettinarip — COMMENTED** [review]
Having this error when I click on the ethereum row
[Screencast from 2024-12-02 09-35-50.webm](https://github.com/user-attachments/assets/8feb0448-d093-4231-94e1-9d896674fdb5)

**Comment by wackerow** [comment]
> Having this error when I click on the ethereum row [Screencast from 2024-12-02 09-35-50.webm](https://github.com/user-attachments/assets/8feb0448-d093-4231-94e1-9d896674fdb5)

Confirming the same

**Comment by nloureiro** [comment]
@corwintines I got this error on ethereum line on the table. 

https://github.com/user-attachments/assets/27b8fa4c-68cc-4bda-84e1-b1f9a5fb196c

**Comment by corwintines** [comment]
@nloureiro @pettinarip @wackerow fixed that issue now :)

**Comment by nloureiro** [comment]
The robust filter is not working

https://github.com/user-attachments/assets/112f6f8c-b872-40a3-b69f-55a0cef88ba1

**Comment by pettinarip** [comment]
Small issue I found when playing with the networks table.

1. expand a row
2. change the filters
3. a different row will be display as expanded
[Screencast from 2024-12-04 17-22-42.webm](https://github.com/user-attachments/assets/6af5e91b-b43c-4418-8591-dae47cd8d7cb)

IMO we should do one of the following:
- close all the expanded rows
- or keep expanded the rows that were expanded (we might need to have an id per row)

**Comment by pettinarip** [comment]
We are missing one translation in the quiz label
image

**Comment by pettinarip** [comment]
Not sure if this is on purpose or not, but I think we are missing the Filters button in mobile

image

while in the wallets table we display it next to the results counter
image

**Comment by corwintines** [comment]
As per conversation with @pettinarip and @konopkja in standup today, going to bring this in

**Pull request merged**